wave_lang/kernel/wave/mlir_converter/water_emitter.py

```diff
@@ -562,8 +581,9 @@ def _emit_from_captured_trace(
             "Transform dialect pipelines are not implemented yet."
         )
 
-    # TODO: Forward locations properly
-    with ir.Context() as ctx, ir.Location.unknown():
+    with ir.Context() as ctx, (
+        trace.location.to_water() if trace.location else ir.Location.unknown()
```

Comment:
Can we have some sort of warning if `trace.location is None` here but location capturing is enabled?